packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -679,19 +1024,19 @@ impl<'m> RunnerImpl<'m> {
     /// TODO: DOC, flushing from a single column
     fn array_data_from_js(
         &mut self,
-        mv8: &'m MiniV8,
-        data: &mv8::Value<'m>,
+        scope: &mut v8::HandleScope<'s>,
+        data: v8::Local<'s, v8::Value>,
         data_type: &DataType,
         len: Option<usize>,
-    ) -> Result<ArrayData> {
+    ) -> JSRunnerResult<ArrayData> {
         // `data` must not be dropped until flush is over, because
         // pointers returned from FFI point inside `data`'s ArrayBuffers' memory.
-        let obj = data.as_object().ok_or_else(|| {
-            Error::Embedded(format!("Flush data not object for field {:?}", data_type))
+        let obj = data.to_object(scope).ok_or_else(|| {
+            JSRunnerError::Embedded(format!("Flush data not object for field {:?}", data_type))
```

Comment:
```suggestion
            JSRunnerError::Embedded(format!("Flush data not object for field {data_type:?}"))
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/02af15786b564ada4f9992194281ff0a3281afb9